src/components/GlobalFilter/TagsModal.tsx

```diff
@@ -23,13 +30,16 @@ export type TagsModalProps = {
 
 export type IDMapper = (tag: CommonTag) => string;
 export type CellsMapper = (tag: CommonTag) => (string | number | boolean | undefined)[];
-export type DebounceCallback = (filters?: TagFilterOptions, pagination?: TagPagination) => Action;
+export type DebounceCallback = (filters?: TagFilterOptions, pagination?: TagPagination) => void | Promise<any>;
 
-export const useMetaSelector = (key: 'tags' | 'workloads' | 'sid') =>
-  useSelector<ReduxState, [boolean | unknown, number, number, number]>(({ globalFilter }) => {
-    const selected = globalFilter[key];
-    return [selected?.isLoaded, selected?.total || 0, selected?.page || 1, selected?.perPage || 10];
-  }, shallowEqual);
+export const useMetaSelector = (key: 'tags' | 'workloads' | 'sid') => {
+  const tags = useAtomValue(tagsAtom);
+  const workloads = useAtomValue(workloadsAtom);
+  const sids = useAtomValue(sidsAtom);
+  const selected = { tags, workloads, sid: sids }[key];
+
+  return [selected?.isLoaded ?? false, selected?.total ?? 0, selected?.page ?? 1, selected?.perPage ?? 10] as [boolean, number, number, number];
```

Comment:
Instead of typecasting the return value, you can define the return value of the function. Its a bit cleaner:

```tsx
export const useMetaSelector = (key: 'tags' | 'workloads' | 'sid'): [boolean, number, number, number]
```

This will ensure that TS will not complain down bellow. and you wont need type casting 🙂